Creation of image prune CronJob template

jefkel opened the pull request:
oc cronjob with cron script in a configMap.
Script will run a soft registry prune followed by a rollout of the docker-registry.

sbarre-esit commented on monitoring/openshift/cronjob-registry-prune.yaml:
```diff
@@ -0,0 +1,150 @@
+  - name: "NAMESPACE"
+    displayName: "Namespace"
+    description: "Name of the Namespace where to deploy the Scheduled Job"
+    value: "cluster-ops"
```
Suggest `advsol-ops`

sbarre-esit commented on monitoring/openshift/cronjob-registry-prune.yaml:
```diff
@@ -0,0 +1,150 @@
+                name: registry-prune-volume
+            restartPolicy: "Never"
+            terminationGracePeriodSeconds: 30
+            activeDeadlineSeconds: 500
```
Is this the time limit for the Job to complete?

jefkel replied:
yes, activeDealineSeconds is the job completion time limit (~8min),  if you have some measurements for job duration that are more appropriate, please feel free to update.

sbarre-esit replied:
Looks like PROD usually finishes in under 5 min
```
# ll /var/log/oc-prune/*.err | awk '{print $8}' | grep ^07 | sort -u
07:00
07:01
07:02
07:03
07:04
```

jefkel replied:
going to bump to 30 minutes (1600s) for now as the job will also stay active to log the status of the docker-registry rollout.

sbarre-esit commented on monitoring/openshift/cronjob-registry-prune.yaml:
```diff
@@ -0,0 +1,150 @@
+          spec:
+            containers:
+              - name: "${JOB_NAME}"
+                image: "openshift/jenkins-slave-base-centos7"
```
`openshift3/ose:v3.10.34` seems to work as well. Not sure we need a pod with java? Think its worth locking this down to match the installed OCP version?

jefkel commented on monitoring/openshift/cronjob-registry-prune.yaml:
```diff
@@ -0,0 +1,150 @@
+  data:
+    ${JOB_SOURCE_SCRIPT}: |
+      NOOP=echo
+      echo "running: oc adm prune images --keep-younger-than=96h"; date
```
You can currently directly edit the configMap that contains the value for any testing updates (we can look at making the bash script a bit easier to maintain inside the configMap)

For a more permanent update, I would suggest that we want the parameter in this version-tracked template to accurately reflect the deployed configuration?

jefkel commented:
Updated with Persistent storage configuration
Added Build and Deployment pruning
README.md updates to reflect deployment.

Deployed to lab for testing - current schedule is daily at 5:00AM

jefkel commented:
I've added the hard prune (noop version) and refactored the object files to be easier to install/manage changes.  Please have another look.

sbarre-esit commented:
Is this a blocking operation?
```
oc rollout status dc/docker-registry -n default
```
or will it just print the current status (say mid rollout) and continue on?

sbarre-esit commented:
Is the DashAI alerting working in PROD yet?

jefkel commented:
> Is this a blocking operation?
> ```
> oc rollout status dc/docker-registry -n default
> ```
This is a blocking command that will watch and print the rollout status messages to stdout until the rollout has either completed or timed out.

> Is the DashAI alerting working in PROD yet?
Looks like something is started for VictorOps, but no Slack notification channels have been configured.

stewartshea commented:
Victorops is an Arctiq tool, so that wouldn't suffice in this case. A slack
channel should be configured.

On Sun, Dec 9, 2018, 12:53 PM Jeff <notifications@github.com> wrote:

> Is this a blocking operation?
>
> oc rollout status dc/docker-registry -n default
>
> This is a blocking command that will watch and print the rollout status
> messages to stdout until the rollout has either completed or timed out.
>
> Is the DashAI alerting working in PROD yet?
> Looks like something is started for VictorOps, but no Slack notification
> channels have been configured.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BCDevOps/openshift-tools/pull/42#issuecomment-445570724>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARdzuLsDQja3amdFykr7GP72fG9s8jeVks5u3XhQgaJpZM4Yv-Tf>
> .
>
-- 

Shea Stewart, Partner
Arctiq: Intelligent Architecture
shea.stewart@arctiq.ca
+1-647-972-5191
http://www.arctiq.ca
